### PR TITLE
More e2e tests

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: adds cluster definitions for Jenkins ginkgo tests to test disabling specific features

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: examples/e2e-tests/kubernetes/release/features-off/addons-disabled.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          },
+          {
+            "name": "rescheduler",
+            "enabled": false
+          }
+        ]
+      }
+    },
+    "masterProfile": {
+      "count": 5,
```
Comment (jackfrancis) [inline examples/e2e-tests/kubernetes/release/features-off/addons-disabled.json]: For these tests let's build 1-master clusters to optimize test time and infra overhead.

File: examples/e2e-tests/kubernetes/kubernetes-config/addons-disabled.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          },
+          {
+            "name": "rescheduler",
+            "enabled": false
+          }
+        ]
+      }
+    },
+    "masterProfile": {
+      "count": 5,
```
Comment (jackfrancis) [inline examples/e2e-tests/kubernetes/kubernetes-config/addons-disabled.json]: I still think we should do single masters for these tests, thoughts?

**Comment by jackfrancis** [comment]
So, I wonder if we should just get rid of `examples/e2e-tests/kubernetes/kubernetes-config/addons-enabled.json` and move that coverage to the standard test (i.e., all addons are enabled for our standard version tests)

File: examples/e2e-tests/kubernetes/kubernetes-config/addons-disabled.json (hunk 2 above)
Comment (CecileRobertMichon) [reply examples/e2e-tests/kubernetes/kubernetes-config/addons-disabled.json]: Yes that's a miss, I'll fix it! same for addons-enabled

**Comment by CecileRobertMichon** [comment]
My reasoning on this was that the standard test should test the default behavior with no optional features enabled/disabled and then we would turn on/off specific features like addons in separate tests. Otherwise what justifies having addons enabled in the standard tests but not other things like extensions and other non-default features?

**Comment by jackfrancis** [comment]
I think "other things like extensions and other non-default features" should also be turned on on the standard tests as you point out. Thoughts?

**Comment by CecileRobertMichon** [comment]
I like that idea but what about testing a "vanilla" cluster with default values for everything? Because the standard test definition is the one used for PR e2e with every k8s version. Also adds the question of, is everything supported in every version?

**Comment by jackfrancis** [comment]
How about this:

- a "vanilla" cluster for each version (different versions will have different default configurations when not user-provided)
- a "kitchen sink" cluster that includes everything that is supported by all versions (configuration options that are only compatible for particular versions get their own test tied to the most recent compatible version)

In this case "kitchen sink" is the definition that we currently use, that we can easily substitute a k8s version into.

**Pull request merged**